### ISSUE TITLE
Irregular durations are not supported (by design?)

### ISSUE THREAD

**Issue by jscottsf** [esnet/pond#117]
From the duration code, it seems that irregular rollups are not supported (or maybe I'm being a bit naive).

The example shows:

```
     * const p = duration({
     *     seconds: 2,
     *     minutes: 2,
     *     hours: 2,
     *     days: 2,
     *     weeks: 2,
     *     months: 2,
     *     years: 2
     * });
```

Maybe months and years should be removed since it can be misleading?

**Comment by sartaj10** [comment]
Similar to the other issue, you can definitely create durations similar to a moment object.

```
it("can construct one with a moment style object", () => {
        const p = duration({
            seconds: 2,
            minutes: 2,
            hours: 2,
            days: 2,
            weeks: 2,
            months: 2,
            years: 2
        });
        expect(+p).toBe(69732122000);
    });
```

**Comment by jscottsf** [comment]
Correct. But a duration created via a moment is normalized to milliseconds. So in essence, it becomes a fixed interval of 30 days. For irregular time intervals (such as months), this does not produce a correct rollup. Incidentally, InfluxDB has a similar limitation.

So maybe it should just be indicated in the docs (or maybe it's obvious), that rollups will not rollup in a proper (irregularly sized) month bin.

**Comment by pjm17971** [comment]
Yeah, I think this is a documentation issue mostly, though irregular rollups used to be part of the library but when it was ported to Typescript it was removed because we didn't use them (and the assumption was that such large irregular periods would be better handled elsewhere).

What are you trying to achieve? Monthly rollups?
